### PR TITLE
STY: pyproject, updates from missions

### LINKED ISSUE #3: Update templates based on other packages
- [x] pysatCDAAC
- [x] pysatMadrigal
- [x] pysatMissions
- [x] pysatModels
- [x] pysatNASA
- [x] pysatSeasons
- [x] pysatSpaceWeather
- [x] pysat

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses https://github.com/pysat/pysatMissions/issues/104

Includes pyproject updates based on pysatNASA

# Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

inspection

# Checklist:

- [ ] Make sure you are merging into the ``develop`` (not ``main``) branch
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Add a note to ``CHANGELOG.md``, summarizing the changes
- [ ] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: .github/workflows/pip_rc_install.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# This workflow will install Python dependencies and the latest RC of pysatNASA from test pypi.
+# This test should be manually run before an RC is officially approved and versioned.
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Test install of latest RC from pip
+
+# Uncomment the "on" statement to activate on Github Actions
```
Comment (aburrell) [inline .github/workflows/pip_rc_install.yml]:
```suggestion
# Uncomment the "on" statement to activate on GitHub Actions
```

File: .github/workflows/pip_rc_install.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# on: [workflow_dispatch]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
+        python-version: ["3.10"]  # Keep this version at the highest supported Python version
```
Comment (aburrell) [inline .github/workflows/pip_rc_install.yml]:
```suggestion
        python-version: ["3.11"]  # Keep this version at the highest supported Python version
```
?

File: .github/workflows/pysat_rc.yml [hunk 3]
```diff
@@ -44,9 +43,9 @@ jobs:
 
     # Replace PACKAGENAME with package name.
```
Comment (aburrell) [inline .github/workflows/pysat_rc.yml]:
```suggestion
```

File: docs/conf.py [hunk 4]
```diff
@@ -20,9 +20,29 @@
 
 import json
 import os
+from pyproject_parser import PyProject
 import sys
 sys.path.insert(0, os.path.abspath('..'))
 
+# -- Project information -----------------------------------------------------
```
Comment (aburrell) [inline docs/conf.py]:
```suggestion
# -- Project information - Replace PACKAGENAME, ETC with appropriate package information ---
```

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+  "Topic :: Scientific/Engineering :: Physics",
+  "Topic :: Scientific/Engineering :: Atmospheric Science",
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: BSD License",
+  "Natural Language :: English",
+  "Programming Language :: Python :: 3",
+  "Programming Language :: Python :: 3.6",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
```
Comment (aburrell) [inline pyproject.toml:27]:
```suggestion
  "Programming Language :: Python :: 3.10",
  "Programming Language :: Python :: 3.11",
```

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  "Operating System :: Microsoft :: Windows"
+]
+keywords = [
+  "pysat"
+]
+dependencies = [
+  "numpy",
+  "packaging",
+  "pandas",
+  "pysat>=3.0.4",
```
Comment (aburrell) [inline pyproject.toml]:
```suggestion
  "pysat>=3.1.0",
```

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  "xarray"]
+
+[project.optional-dependencies]
+aacgmv2 = ["aacgmv2"]
+apexpy = ["apexpy"]
+OMMBV = ["OMMBV"]
```
Comment (aburrell) [inline pyproject.toml]: These shouldn't be here. Maybe turn into a comment?

File: setup.cfg [hunk 8]
```diff
@@ -1,54 +1,11 @@
+# name and version must be maintained here as well for python 3.6 compatibility
```
Comment (aburrell) [inline setup.cfg]:
```suggestion
# Name and version must be maintained here as well for Python 3.6 compatibility
```

File: pyproject.toml (hunk 7 above)
Comment (jklenzing) [inline pyproject.toml]:
```suggestion
[project.optional-dependencies]
```

File: .github/workflows/pip_rc_install.yml [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install standard dependencies
+      run: pip install -r requirements.txt
+
+    - name: Install pysatNASA RC
```
Comment (aburrell) [inline .github/workflows/pip_rc_install.yml]:
```suggestion
      - name: Install PACKAGENAME RC
```

File: .github/workflows/pip_rc_install.yml [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+    - name: Set up pysat
+      run: |
+        mkdir pysatData
+        python -c "import pysat; pysat.params['data_dirs'] = 'pysatData'"
+
+    - name: Check that install imports correctly
+      run: |
+        cd ..
+        python -c "import pysatNASA; print(pysatNASA.__version__)"
```
Comment (aburrell) [inline .github/workflows/pip_rc_install.yml]:
```suggestion
        python -c "import PACKAGENAME; print(PACKAGENAME.__version__)"
```

File: pyproject.toml [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+description = 'pysat support for DATA OR ANALYSIS'
+readme = "README.md"
+requires-python = ">=3.6"
+license = {file = "LICENSE"}
+authors = [
+    {name = "AUTHOR", email = "pysat.developers@gmail.com"}
+]
+keywords =
+    pysat
+classifiers =
```
Comment (jklenzing) [inline pyproject.toml]:
```suggestion
```

File: .github/workflows/pip_rc_install.yml [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install standard dependencies
+      run: pip install -r requirements.txt
+
+      - name: Install PACKAGENAME RC
+      run: pip install --no-deps --pre -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ pysatNASA
```
Comment (aburrell) [inline .github/workflows/pip_rc_install.yml]:
```suggestion
      run: pip install --no-deps --pre -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ PACKAGENAME
```

Given I've been spotty about finding instances of 'pysatNASA' throughout, I'd recommend doing a grep for it in these files and whatever is in the Missions package.

**Comment by jklenzing** [comment]
pinging @aburrell for re-review

File: .github/workflows/pip_rc_install.yml (hunk 12 above)
Comment (jklenzing) [reply .github/workflows/pip_rc_install.yml]: Did a local search for both nasa and missions, this was the last one to be fixed

**Review by aburrell — APPROVED** [review]

**Pull request merged**